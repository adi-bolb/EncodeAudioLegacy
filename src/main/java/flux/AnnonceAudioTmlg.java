package flux;

import java.io.ByteArrayInputStream;
import java.io.IOException;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.xpath.XPath;
import javax.xml.xpath.XPathConstants;
import javax.xml.xpath.XPathExpressionException;
import javax.xml.xpath.XPathFactory;

import org.w3c.dom.Document;
import org.w3c.dom.Node;
import org.xml.sax.SAXException;

public final class AnnonceAudioTmlg implements IAnnonceAudioTmlg {
	private String url;
	private String format;
	private String fileName;

	public AnnonceAudioTmlg(String message) throws ParserConfigurationException, SAXException, IOException, XPathExpressionException {
		DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
		Document document = builder.parse(new ByteArrayInputStream(message.getBytes()) );
		XPathFactory newInstance = XPathFactory.newInstance();
		XPath xPath = newInstance.newXPath();
		Node annonceNode = (Node) xPath.evaluate("//obj[@name='annonceAudio']", document, XPathConstants.NODE);
		fileName = getValOffAttribute(xPath, annonceNode, "fileName");
		url = getValOffAttribute(xPath, annonceNode, "url");
		format = getValOffAttribute(xPath, annonceNode, "format");
	}

	private static String getValOffAttribute(XPath xPath, Node annonceNode,
			String attribute) throws XPathExpressionException {
		return xPath.evaluate("*[@name='ATTR']/@val".replace("ATTR", attribute), annonceNode);
	}


	public void setUrl(String string) {
		url = string;

	}

	public void setFormat(String format) {
		this.format = format;
	}

	public void setFileName(String name) {
		fileName = name;
	}

	public String getUrl() {
		return url;
	}

	public String getFileName() {
		return fileName;
	}

	@Override
	public String toString() {
		return "AnnonceAudioTmlg [url=" + url + ", format=" + format
				+ ", fileName=" + fileName + "]\n";
	}

	public String getFormat() {
		return format;
	}


}