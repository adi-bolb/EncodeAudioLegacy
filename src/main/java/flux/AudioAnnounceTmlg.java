package flux;

import java.io.ByteArrayInputStream;
import java.io.IOException;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.xpath.XPath;
import javax.xml.xpath.XPathConstants;
import javax.xml.xpath.XPathExpressionException;
import javax.xml.xpath.XPathFactory;

import org.w3c.dom.Document;
import org.w3c.dom.Node;
import org.xml.sax.SAXException;

public final class AudioAnnounceTmlg implements IAudioAnnounceTmlg {
	private String url;
	private String format;
	private String fileName;

	public AudioAnnounceTmlg(String message) throws ParserConfigurationException, SAXException, IOException, XPathExpressionException {
		DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
		Document document = builder.parse(new ByteArrayInputStream(message.getBytes()));
		XPathFactory newInstance = XPathFactory.newInstance();
		XPath xPath = newInstance.newXPath();
		Node announceNode = (Node) xPath.evaluate("//obj[@name='annonceAudio']", document, XPathConstants.NODE);
		fileName = getValOffAttribute(xPath, announceNode, "fileName");
		url = getValOffAttribute(xPath, announceNode, "url");
		format = getValOffAttribute(xPath, announceNode, "format");
	}

	private static String getValOffAttribute(XPath xPath, Node announceNode,
			String attribute) throws XPathExpressionException {
		return xPath.evaluate("*[@name='ATTR']/@val".replace("ATTR", attribute), announceNode);
	}


	public void setUrl(String string) {
		url = string;

	}

	public void setFormat(String format) {
		this.format = format;
	}

	public void setFileName(String name) {
		fileName = name;
	}

	public String getUrl() {
		return url;
	}

	public String getFileName() {
		return fileName;
	}


	public String getFormat() {
		return format;
	}

	@Override
	public String toString() {
		return "AnnounceAudioTmlg [url=" + url + ", format=" + format
				+ ", fileName=" + fileName + "]";
	}
}