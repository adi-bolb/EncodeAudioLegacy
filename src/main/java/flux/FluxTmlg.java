package flux;

public class FluxTmlg implements IFluxTmlg {

	private IBodyTmlg body;

	public FluxTmlg(AudioAnnounceTmlg audioAnnounceTmlg) {

			final IAudioAnnounceTmlg audioAnnounce = new AudioAnnounceTmlg(audioAnnounceTmlg);
			final ITravelInfoTmlg travelInfo = new ITravelInfoTmlg() {

				public IAudioAnnounceTmlg getAudioAnnounce() {
					return audioAnnounce;
				}
			};
			body = new IBodyTmlg() {

				public ITravelInfoTmlg getTravelInfo() {
					return travelInfo;
				}

				@Override
				public String toString() {
					return audioAnnounce.toString();
				}
			};
	}

	public IBodyTmlg getBody() {
		return body;
	}

	@Override
	public String toString() {
		return "FluxTmlg [body=" + body + "]";
	}
}