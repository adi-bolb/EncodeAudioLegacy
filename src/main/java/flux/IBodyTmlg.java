package flux;

public interface IBodyTmlg {

	ITravelInfoTmlg getTravelInfo();

}
