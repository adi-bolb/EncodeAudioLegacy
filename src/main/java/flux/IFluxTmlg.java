package flux;

public interface IFluxTmlg {

	IBodyTmlg getBody();

}
