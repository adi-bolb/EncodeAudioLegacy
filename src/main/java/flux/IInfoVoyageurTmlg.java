package flux;


public interface IInfoVoyageurTmlg {

	IAnnonceAudioTmlg getAnnonceAudio();

}
