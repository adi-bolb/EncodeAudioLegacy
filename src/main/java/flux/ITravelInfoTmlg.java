package flux;


public interface ITravelInfoTmlg {

	IAudioAnnounceTmlg getAudioAnnounce();

}
