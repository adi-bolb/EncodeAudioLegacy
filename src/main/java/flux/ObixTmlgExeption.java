package flux;

public class ObixTmlgExeption extends Exception {

}
