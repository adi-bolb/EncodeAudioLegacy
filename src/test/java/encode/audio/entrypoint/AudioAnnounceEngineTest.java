package encode.audio.entrypoint;

import org.approvaltests.Approvals;
import org.approvaltests.legacycode.LegacyApprovals;
import org.junit.Ignore;
import org.junit.Test;

import flux.AudioAnnounceTmlg;

public class AudioAnnounceEngineTest {

	private static final String REMOTE_AUDIO_FILE_NAME = "10.151.156.180Mon_Nov_04_140724_CET_2013343.wav";
    
	@Test public void 
    coverageAudioAnnounceEnginLockdown() throws Exception {
         Object[] sourceFileNames = {REMOTE_AUDIO_FILE_NAME, "10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3"};
        Object[] targetFormats = {"wav", "mp3", "ogg"};
        Object[] finalUrls = { "null10.151.156.180Mon_Nov_04_140724_CET_2013343", "null10.151.156.180Tue_Nov_05_141112_CET_2013343"} ;
        LegacyApprovals.LockDown(this, "publishAudioFileVariations", targetFormats, finalUrls, sourceFileNames);
    }
    
    public String publishAudioFileVariations(String targetFormat, String finalUrl, String sourceFileName) throws AppTechnicalException {
        // Given
        AudioAnnounceTmlg audioFileMessage = new AudioAnnounceTmlg(finalUrl, targetFormat, sourceFileName);
        DataObject configAudioTmp = new AudioDataObject("." + targetFormat);
        DataObject httpDataObj = new HttpDataObj("./src/test/resources/", "http://localhost/get");
        
        LocalHTTPSServer localServerFolder = new LocalHTTPSServer();
        LocalTmpFolder localTmpFolder = new LocalTmpFolder();
        AudioAnnounceEngine audioAnnounceEngine = new AudioAnnounceEngine(localServerFolder, localTmpFolder);

        // When
        return  audioAnnounceEngine.publishAudioFile(audioFileMessage, configAudioTmp, httpDataObj);

    } 
    
}
