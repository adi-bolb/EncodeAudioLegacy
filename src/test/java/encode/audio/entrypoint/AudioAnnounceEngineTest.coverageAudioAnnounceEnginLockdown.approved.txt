[wav, null10.151.156.180Mon_Nov_04_140724_CET_2013343, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = java.lang.reflect.InvocationTargetException 
[mp3, null10.151.156.180Mon_Nov_04_140724_CET_2013343, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[ogg, null10.151.156.180Mon_Nov_04_140724_CET_2013343, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = java.lang.reflect.InvocationTargetException 
[wav, null10.151.156.180Tue_Nov_05_141112_CET_2013343, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = java.lang.reflect.InvocationTargetException 
[mp3, null10.151.156.180Tue_Nov_05_141112_CET_2013343, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3, format=mp3, fileName=10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3]] 
[ogg, null10.151.156.180Tue_Nov_05_141112_CET_2013343, 10.151.156.180Mon_Nov_04_140724_CET_2013343.wav] = java.lang.reflect.InvocationTargetException 
[wav, null10.151.156.180Mon_Nov_04_140724_CET_2013343, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Tue_Nov_05_141112_CET_2013343.wav, format=wav, fileName=10.151.156.180Tue_Nov_05_141112_CET_2013343.wav]] 
[mp3, null10.151.156.180Mon_Nov_04_140724_CET_2013343, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = java.lang.reflect.InvocationTargetException 
[ogg, null10.151.156.180Mon_Nov_04_140724_CET_2013343, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = java.lang.reflect.InvocationTargetException 
[wav, null10.151.156.180Tue_Nov_05_141112_CET_2013343, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = FluxTmlg [body=AnnounceAudioTmlg [url=http://localhost/get10.151.156.180Tue_Nov_05_141112_CET_2013343.wav, format=wav, fileName=10.151.156.180Tue_Nov_05_141112_CET_2013343.wav]] 
[mp3, null10.151.156.180Tue_Nov_05_141112_CET_2013343, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = java.lang.reflect.InvocationTargetException 
[ogg, null10.151.156.180Tue_Nov_05_141112_CET_2013343, 10.151.156.180Tue_Nov_05_141112_CET_2013343.mp3] = java.lang.reflect.InvocationTargetException 
